Automattic/wp-calypso
# Domains: Add titan email upsell page contents

Related to #54941 

#### Changes proposed in this Pull Request

Adds email upsell content to the new page setup in #54941 reusing the existing `<EmailProvidersComparison>` component. ~~Also incorporates the component design changes for the former component~~. The page can be viewed at:

```
http://wordpress.com/domains/add/:domain/email/:siteName
```

Changes I wanted to call out:

1. Refactored `EmailProvidersComparison` component to support props to inject text content for the Domains page.
2. ~~Extracted out `TitanNewMailbox` and `TitanNewMailboxList` components to fix styling issues since styles were previously clubbed inside `TitanAddMailboxes` component.~~ See #55249
3. ~~Modified `HeaderCakeBack` to support `iconPosition` for the _Skip_ button (not sure if we're going ahead with this)~~
4. ~~Removed the old trash button next to input field and added the _Remove this mailbox_ button adjacent to _Add another mailbox_ button, along with layout changes.~~ See #55529

#### Testing instructions
<details>
<summary>Click to expand</summary>


1. Run git checkout `add/titan-email-upsell-contents` and start dev server
2. Log into a WordPress.com account that has a site
3. Open the _Domains_ page (replace `yoursite` with a site slug that you have): `http://calypso.localhost:3000/domains/manage/yoursite.wordpress.com`
4. Click "Just search for a domain" button
5. Search for a domain or choose one of the suggestions by clicking _Select button_
5. Once the page loads, replace `google-workspace` in the domain with `email`. For example:
    If the URL shown is:
    ```
    http://calypso.localhost:3000/domains/add/example.in/google-workspace/example.wordpress.com
    ```
   
    replace it with:

    ```
    http://calypso.localhost:3000/domains/add/example.in/email/example.wordpress.com
    ```
6. Verify that the new route loads and renders without errors

Things to consider:

1. Need to verify the changes to `EmalProvidersComparison` and related components don't break/introduce issues in other areas like the _Emails_ section
2. ~~Layout changes adapt well in mobile/desktop layouts~~
</details>

#### Screenshots/Recording

<details>
<summary>Click to expand</summary>

##### Recording
_P.S. ignore the 1Password popover_

https://user-images.githubusercontent.com/2759499/129753672-b190ac28-7480-4f80-9ec2-b577368150b5.mp4

##### Screenshots
![dp-flow__google__desktop](https://user-images.githubusercontent.com/2759499/129753715-7dd58f33-78ee-4d17-921c-d04ebf679d31.jpeg)
![dp-flow__titan__desktop](https://user-images.githubusercontent.com/2759499/129753717-7d2e0ce7-8a77-4776-a6c3-9b42014f35e4.jpeg)
![dp-flow__mobile](https://user-images.githubusercontent.com/2759499/129753728-6dae2420-018c-42b8-85ad-4825aeec4bf5.jpeg)
</details>

---

#### TODO
- [x] Get feedback on the current approach
- [x] Fix the broken Google Workspace card
- [x] Fix the broken _Add Email_ functionality
